src/physics/smoothing.cxx

```diff
@@ -319,12 +320,17 @@ BoutReal Average_XY(const Field2D &var) {
   return Vol_Glb;
 }
 
+BoutReal Average_XY(const Field3D&) {
+  AUTO_TRACE();
+  throw BoutException("Average_XY(Field3D) not yet implemented.");
+}
```

Comment:
Why are we adding a function just to say it's not implemented?

Reply:
This is because `Vol_Integral` calls `Average_XY(<metric type>)`, so will fail to compile without the implementation. This looks like the only place it's called, so I've changed it to throw in `Vol_Integral` directly.